src/library/scala/collection/immutable/NumericRange.scala

```diff
@@ -118,16 +120,64 @@ sealed class NumericRange[T](
     }
   }
 
+  private def indexOfTyped(elem: T, from: Int): Int = {
+    val pos = posOf(elem)
+    pos match {
+      case -2                 => super.indexOf(elem, from)
+      case pos if pos >= from => pos
+      case _                  => -1
+    }
+  }
+
+  final override def indexOf[B >: T](elem: B, from: Int): Int = {
+    try indexOfTyped(elem.asInstanceOf[T], from)
+    catch { case _: ClassCastException => super.indexOf(elem, from) }
+  }
```

Comment:
lose braces as with lastIndexOf. Worth adding that some people like extra brackets if there is a risk of ambiguity or orphan expressions.